core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -1731,6 +1756,70 @@ private ExprValue exprYear(ExprValue date) {
     return new ExprIntegerValue(date.dateValue().getYear());
   }
 
+  /**
+   * Convert mode argument passed into our CalendarLookup class to a different mode.
+   * Needed to align with MySQL for the yearweek function due to different behaviour for modes.
+   * Note, this misalignment only exists for yearweek.
+   * Our current mode behavior works as intended for other functions.
+   *
+   * @param mode is an integer containing the initial mode arg
+   * @return an integer containing the new mode
+   */
+  public int convertWeekModeFromMySqlToJava(LocalDate date, int mode) {
+    // Needed to align with MySQL. Due to how modes for this function work.
+    // See description of modes here ...
+    // https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_week
+
+    if ((0 <= mode && mode <= 4) && (CalendarLookup.getWeekNumber(mode, date) == 0)) {
+      mode = 2;
+    } else if ((mode == 5) && (CalendarLookup.getWeekNumber(mode, date) == 0)) {
```

Comment:
you can combine the `CalendarLookup.getWeekNumber` calls:
```
if (CalendarLookup.getWeekNumber() == 0) {
  switch (mode) {
    case 0:
    case 1:
    case 2:
    case 3:
    case 4: 
      return 2;
    case 5:
      return 7;
      break;
    default:  //no-op
  }
  return mode;
}

```

Reply:
Cannot do it this way due to code coverage. Since the minimum number `CalendarLookup.getWeekNumber()` returns is 1 for case 6 and 7 (and also 1 which is a bug in my original code) it will never enter the if-statement and the default case is never checked.

Reply:
Other options: 
```
      if (CalendarLookup.getWeekNumber() != 0) {
        return mode;
      }
      if (mode <= 4) {
        return 2;
      }
      return 7;
```

```
return CalendarLookup.getWeekNumber() != 0 ? mode : 
           mode <= 4 ? 2 : 
           7;
```

Reply:
Fixed in https://github.com/Bit-Quill/opensearch-project-sql/pull/236/commits/271bf2dba56d0537c6223e3b888345d2738720ee (Thanks @acarbonetto!)